conf/config.go

```diff
@@ -190,3 +216,102 @@ func UnsetEnv(protect *testing.T, key string) error {
 	return err
 
 }
+
+/***********************************************************************************************
+
+    CODE BELOW THIS BLOCK IS EXPERIMENTAL AND ONLY BEING USED INTERNALLY BY CONF PACKAGE
+
+***********************************************************************************************/
+
+// Gopath function exposes the gopath of the application while keeping it immutable. Golang does
+// not allow const to be strings, and a var would make it mutable if made public. This could be
+// something the config / viper struct keeps track. For now it's separate.
+func Gopath() string {
+	return envVars.gopath
+}
+
+/*
+Checkout function takes a reference to a struct or a slice of string. It will traverse both
+data structures and look up key value pairs in the conf / viper struct by the name of the field
+for structs and string values for the slice. The function works with pointers so no value is
+returned. An error is returned if the wrong data structure is provided.
+*/
+func Checkout(list interface{}) error {
+
+	// Check if the data type provided is supported
+	switch list := list.(type) {
+	// If it's a slice of strings
+	case []string:
+
+		for n, str := range list {
+			if val, exists := LookupEnv(str); exists {
+				list[n] = val
+			}
+		}
+
+		return nil
+	// Checking the rest of data types through reflection
+	default:
+		// Get the concrete value from the interface
+		check := reflect.ValueOf(list)
+		// Is it a pointer?
+		if check.Kind() == reflect.Ptr {
+
+			// Dereference the pointer
+			el := check.Elem()
+
+			// Is the data type a struct?
+			if el.Kind() == reflect.Struct {
+
+				// Get ready to walk the fields of the structs
+				num := el.NumField() // Number of fields in struct
+				elType := el.Type()  // Used for tags
```

Comment:
If we know it's a struct - can we call the `walk` method directly instead of having the parsing logic here?

Reply:
Yes!